query_postal_codes_heatmaps_api.py

```diff
@@ -0,0 +1,103 @@
+import random
+from collections import defaultdict
+import statistics
+
+from datetime import datetime, timedelta, date
+import matplotlib.pyplot as plt
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+
+def build_api_query(date: date, tile_ids: list) -> str:
+    api_request = (
+        BASE_URL
+        + "/heatmaps/dwell-density/daily/{0}".format(date.isoformat())
+        + "?tiles="
+        + "&tiles=".join(map(str, tile_ids))
+    )
+    return api_request
+
+def compute_density_baseline(tile_ids: list, 
+    start_date=date(year=2020, month=1, day=6), nb_days = 29) -> dict:
+    # The baseline is the median value, for the corresponding day of the week,
+    # during the nb_days following start_date"
+    day2densities = defaultdict(list)
+    for delta in range(nb_days):
+        dt = start_date + timedelta(days=delta)
+        day2densities[dt.weekday()].append(get_daily_density(dt, tile_ids))
+    weekday2density = {wday: statistics.median(v)
+                       for wday, v in day2densities.items()}
+    return weekday2density
+
+
+def get_daily_density(date: date, tile_ids: list) -> float:
+    tiles_date = oauth.get(build_api_query(date, tile_ids),
+                           headers=headers).json()['tiles']
+    return sum([t['score'] for t in tiles_date])
+
+
+def get_density_variation(date: date, tile_ids: list, weekday2density: dict) -> float:
+    daily_score_date = get_daily_density(date, tile_ids)
+    density_baseline = weekday2density[date.weekday()]
+    variation = (daily_score_date - density_baseline) / density_baseline
+    return 100*variation
+
+
+def get_tile_ids_postal_code(postal_code: int) -> list:
+    # Randomly sample MAX_NB_TILES_REQUEST tile ids
+    # associated with the postal code of interest
+    muni_tiles_json = oauth.get(
```

Comment:
I guess that "muni" was for municipalities?
```suggestion
    tiles_json = oauth.get(
```

Reply:
Good catch!